core/src/main/scala/kafka/log/OffsetIndex.scala

```diff
@@ -216,26 +216,11 @@ object OffsetIndex extends Logging {
   * for the the broker with a lot of log segments
   *
   */
-class LazyOffsetIndex(@volatile private var _file: File, baseOffset: Long, maxIndexSize: Int = -1, writable: Boolean = true) {
-  @volatile private var offsetIndex: Option[OffsetIndex] = None
-
-  def file: File = {
-    if (offsetIndex.isDefined)
-      offsetIndex.get.file
-    else
-      _file
-  }
+class LazyOffsetIndex(private val _file: File, baseOffset: Long, maxIndexSize: Int = -1, writable: Boolean = true)
```

Comment:
We can remove `private val` from `_file` here and in `LazyTimeIndex`. That's the default for classes.